secure_message/v2/repository/retriever.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+
+from structlog import wrap_logger
+from sqlalchemy import and_, or_
+from werkzeug.exceptions import InternalServerError
+
+from secure_message.repository.database import SecureMessage, Status
+from secure_message.repository.retriever import Retriever
+from secure_message import constants
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class RetrieverV2(Retriever):
+    """Created when retrieving messages"""
+
+    @staticmethod
+    def unread_message_count_by_survey(user, survey):
+        """Count users unread messages for a specific survey"""
+        if user.is_internal:
+            status_conditions, survey_conditions = RetrieverV2._get_conditions_internal_user(survey, user)
+        else:
+            status_conditions, survey_conditions = RetrieverV2._get_conditions_respondent(survey, user)
+
+        try:
+            result = SecureMessage.query.join(Status). \
+                filter(or_(*status_conditions)). \
+                filter(and_(*survey_conditions)). \
+                filter(Status.label == 'UNREAD').count()
+        except Exception as e:
+            logger.error('Error retrieving count of unread messages from database', error=e)
+            raise InternalServerError(description="Error retrieving count of unread messages from database")
+        return result
+
+    @staticmethod
+    def _get_conditions_internal_user(survey, user):
```

Comment:
What do you mean by conditions?  I might've been told and forgotten but I think what these 2 functions do could maybe warrant a comment for clarity

Reply:
Adding comments . These functions set the query conditions/predicates used in the query to get a count of messages when the actor is internal . There is a corresponding one for external . 
Having said that I think I can see a bug .